### PR TITLE
fix: Email validation tokens

### PR DESCRIPTION
**Pull request by benjackwhite**
## Problem

Some investigation around email resetting revealed a few less-than-perfect things. Nothing major from what I can see but confusing APIs that make it look like you can do bad things (which you can't)

## Changes

* Adds some missing tests to verify a bunch of things such as not being able to use someone else's token or re-use a token for verifying a pending email (there were some tests here but not enough to immediately confirm some security doubts)
* Fix an issue with the `login_timestamp` not being included in the token validation (which is what was allowing an email verification to be used multiple times) - this anyway had a 24 hour limit but now it is even more restrictive
* Makes the unauthenticated endpoints non-detail specific which fixes confusion around where the `uuid` was coming from.

👉 _Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review._

### CODE REVIEW

File: frontend/src/scenes/authentication/signup/verify-email/verifyEmailLogic.ts [hunk 1]
```diff
@@ -48,7 +48,7 @@ export const verifyEmailLogic = kea<verifyEmailLogicType>([
             {
                 requestVerificationLink: async ({ uuid }: { uuid: string }) => {
                     try {
-                        await api.create(`api/users/${uuid}/request_email_verification/`, { uuid })
+                        await api.create(`api/users/request_email_verification/`, { uuid })
```
Comment (MichaelKutsch-ph) [inline frontend/src/scenes/authentication/signup/verify-email/verifyEmailLogic.ts:51]: this is breaking currently valid verification links, right? so everyone who created an account and did not verify yet won't be able to verify? Or am I confusing this?

Comment (MichaelKutsch-ph) [reply frontend/src/scenes/authentication/signup/verify-email/verifyEmailLogic.ts:51]: (this comment actually makes more sense in line 34 sry)

Comment (benjackwhite) [reply frontend/src/scenes/authentication/signup/verify-email/verifyEmailLogic.ts:51]: Nope.

**Review by MichaelKutsch-ph — APPROVED** [review]

**Review by zlwaterfield — APPROVED** [review]
This looks good - thanks for making this change.

One thing to note: this will invalidate all existing tokens, which is mildly annoying to users - we may get increased support requests related to this in the next 24 hours.

**Pull request merged**